h2o-core/src/main/java/water/ClearDKVTask.java

```diff
@@ -0,0 +1,106 @@
+package water;
+
+import hex.Model;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.util.Log;
+
+import java.util.*;
+
+/**
+ * Clears keys in all H2O nodes, except for the ones marked as retained.
+ * Only Model and Frame keys are retained. If a key of any other type is provided, it will be removed as well.
+ */
+public class ClearDKVTask extends MRTask<ClearDKVTask> {
+
+  private final Key[] _retainedKeys; // Only model and frame keys
+
+  /**
+   *
+   * @param retainedKeys Keys that are NOT deleted and will remain in DKV. Only Model keys and Frame keys are accepted
+   */
+  public ClearDKVTask(Key[] retainedKeys) {
+    _retainedKeys = retainedKeys;
+  }
+
+  @Override
+  protected void setupLocal() {
+    final Set<Key> retainedKeys = new HashSet<>(_retainedKeys.length);
+    retainedKeys.addAll(Arrays.asList(_retainedKeys));
+    extractNestedKeys(retainedKeys);
+    final Collection<Value> storeKeys = H2O.STORE.values();
+    Futures removalFutures = new Futures();
+    System.out.println(Arrays.toString(retainedKeys.toArray()));
+    for (final Value value : storeKeys) {
+      if (retainedKeys.contains(value._key)) {
+        continue;
+      }
+      if(value.isNull()) continue;
+
+      if (value.isFrame()) {
```

Comment:
Note: Multiple nodes will theoretically call remove operation of one key is on multiple nodes.